apps/communications/contacts/views/details/js/boot.js

```diff
@@ -45,6 +46,62 @@ window.onload = function() {
       document.body.classList.remove('hidden');
     });
 
+    function getContactInfo(id) {
+      return new Promise(function(resolve, reject) {
+        ContactsService.get(params.contact, function onSuccess(savedContact) {
+          ContactsService.getCount(function(count) {
+            resolve(
+              {
+                contact: savedContact,
+                count: count
+              }
+            );
+          });
+        }, function onError() {
+          console.error('Error retrieving contact');
+          reject();
+        });
+      });
+    }
+
+
+    // Get action from URL (new or update)
+    var params = ParamUtils.get();
+    if (params && params.contact) {
+      getContactInfo(params.contact).then(function(info) {
+        DetailsController.setContact(params.contact);
+        DetailsUI.render(info.contact, info.count, false);
+        window.addEventListener(
+          'pageshow',
+          function checkIfUpdate() {
```

Comment:
as well this fucnction could be at the same level than `getContactInfo` to improve readability